var/spack/repos/builtin/packages/alps/package.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+# Refs for building from source and recipes
+# https://alps.comp-phys.org/mediawiki/index.php/Building_ALPS_from_source
+# https://github.com/easybuilders/easybuild-easyconfigs/tree/master/easybuild/easyconfigs/a/ALPS
+# https://github.com/conda-forge/alps-feedstock/tree/master/recipe
+
+
+class Alps(CMakePackage):
+    """Algorithms for Physics Simulations
+
+    Tags: Condensed Matter Physics, Computational Physics
+    """
+
+    homepage = "https://alps.comp-phys.org"
+    url = "https://alps.comp-phys.org/static/software/releases/alps-2.3.0-src.tar.gz"
+
+    version(
+        "20220304_r7871", sha256="74bcb9156701f81439af3c60ecf26afeb6458c48012729aea2e9f7aa34e87426"
+    )
+    # version 2.3.0 is removed since it requires python@:3.6 that's not supported by Spack
+
+    # build fails as of boost@1.83
+    depends_on(
+        "boost@:1.82.0"
+        "+chrono +date_time +filesystem +iostreams +mpi +numpy +program_options"
+        "+python +regex +serialization +system +test +thread +timer"
+    )
+    depends_on("fftw")
+    depends_on("hdf5 ~mpi+hl")
+    depends_on("lapack")
+    depends_on("python", type=("build", "link", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
+
+    extends("python")
+
+    def url_for_version(self, version):
+        if str(version) == "2.3.0":
+            url = "https://alps.comp-phys.org/static/software/releases/alps-2.3.0-src.tar.gz"
```

Comment:
If this is no longer supported, we can simplify this function to work for all versions.